app/scripts/modules/core/src/application/config/managedResources/ManagedResourceConfig.tsx

```diff
@@ -0,0 +1,137 @@
+          Spinnaker is configured to continuously manage some resources, but management has been paused. When resumed,
+          Spinnaker will take action to make each resource match its declarative configuration.
+        </p>
+      </>
+    );
+  } else {
+    return (
+      <>
+        <p>
+          <span className="rainbow-icon">🌈</span> <b>Spinnaker is continuously managing some resources.</b>
+        </p>
+        <p className="sp-margin-l-bottom">
+          If you need to temporarily stop Spinnaker from managing resources — for example, if something is wrong and
+          manual intervention is required — you can pause management and resume it later. Pausing affects all managed
+          resources within this application.
+        </p>
+      </>
+    );
+  }
+};
+
+const ManagedResourceConfig = ({ application }: IManagedResourceConfigProps) => {
+  const [pausePending, setPausePending] = useState(false);
+  const [pauseFailed, setPauseFailed] = useState(false);
+  const [paused, setPaused] = useState(false);
+  const { status: vetoStatus, result: vetos, refresh } = useData(() => ManagedReader.getApplicationVetos(), [], []);
+
+  const isRejected = vetos && vetos.includes(application.name);
+  useEffect(() => {
+    setPaused(isRejected);
+  }, [isRejected]);
+
+  const pauseManagement = () => {
+    setPausePending(true);
+    setPauseFailed(false);
+    logClick('Pause Management', application.name);
+
+    ManagedWriter.pauseResourceManagement(application.name)
+      .then(() => setPaused(true))
+      .catch(() => setPauseFailed(true))
+      .finally(() => setPausePending(false));
+  };
+
+  const resumeManagement = () => {
+    setPausePending(true);
+    setPauseFailed(false);
+    logClick('Resume Management', application.name);
+
+    ManagedWriter.resumeResourceManagement(application.name)
+      .then(() => setPaused(false))
+      .catch(() => setPauseFailed(true))
+      .finally(() => setPausePending(false));
+  };
+
+  if (['NONE', 'PENDING'].includes(vetoStatus)) {
+    return <Spinner size="medium" />;
+  } else if (vetoStatus === 'REJECTED') {
+    return (
+      <div className="alert alert-danger">
+        Something went wrong.{' '}
+        <button className="btn btn-link" onClick={refresh}>
+          Try again
+        </button>
+      </div>
+    );
+  }
+
+  const iconClass = paused ? 'fa-play' : 'fa-pause';
+
+  return (
+    <div className="ManagedResourceConfig">
+      {getManagementStatus(paused)}
+      <button
+        className="btn btn-primary"
+        disabled={pausePending}
+        onClick={paused ? resumeManagement : pauseManagement}
+        type="button"
+      >
+        {(!pausePending && <i className={classNames('fa sp-margin-xs-right', iconClass)} />) || <ButtonBusyIndicator />}{' '}
+        {paused ? 'Resume Management' : 'Pause Management'}
+      </button>
+      {pauseFailed && <div className="alert alert-danger sp-margin-l-top">Saving failed.</div>}
```

Comment:
can this use the `<ValidationMessage />` component? (which I totally want to rename to simply `<Message/>`)

Reply:
44e5e02